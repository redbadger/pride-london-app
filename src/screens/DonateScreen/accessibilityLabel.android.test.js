// @flow
import getAccessibilityLabel from "./accessibilityLabel.android";

const cases = [
  {
    name: "not focused, value",
    values: ["Company", undefined, "Aperture Science", false, undefined]
  },
  {
    name: "not focused, no value",
    values: ["Company", undefined, "", false, undefined]
  },
  {
    name: "focused, value",
    values: ["Company", undefined, "Aperture Science", true, undefined]
  },
  {
    name: "focused, no value",
    values: ["Company", undefined, "", true, undefined]
  }
];

cases.forEach(c => {
  it(`returns value TalkBack would say if: ${c.name}`, () => {
    const output = getAccessibilityLabel(...c.values);
    expect(output).toMatchSnapshot();
  });
});
