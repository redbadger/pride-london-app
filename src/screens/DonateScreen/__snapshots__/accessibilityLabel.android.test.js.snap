// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns value TalkBack would say if: focused, no value 1`] = `"editing, edit box, Company; double tap and hold to long press"`;

exports[`returns value TalkBack would say if: focused, value 1`] = `"editing, Aperture Science, edit box, Company; double tap and hold to long press"`;

exports[`returns value TalkBack would say if: not focused, no value 1`] = `"edit box, Company; double tap to enter text"`;

exports[`returns value TalkBack would say if: not focused, value 1`] = `"Aperture Science, edit box, Company; double tap to enter text"`;
